sklearn/utils/_param_validation.py

```diff
@@ -219,6 +227,9 @@ class StrOptions(_Constraint):
 
     deprecated : set of str or None, default=None
         A subset of the `options` to mark as deprecated in the repr of the constraint.
+
+    internal : set of str or None, default=None
```

Comment:
To be consistent should this be `hidden`?

Reply:
I'm ok with that

Reply:
Acutally, I even think we should remove it and do `[StrOptions({"str1", "str2"}), Hidden(StrOptions({"warn"}))]`

Reply:
I just pushed the change. I find it clearer. Let me know what you think.